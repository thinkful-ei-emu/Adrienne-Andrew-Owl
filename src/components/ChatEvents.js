import React from 'react';
import ChatItem from './ChatItem';

function ChatEvents(props) {
  
  
  const chatItems = props.chatEvents.map(chatEvent => {
    // let lastParticipant = chatEvent.participantId;
    let participant = props.participants.find(participant => chatEvent.participantId === participant.id);

    return (
      <ChatItem
        key={chatEvent.timestamp}
        chatEvent={chatEvent}
        participant={participant} 
      />
    )
  });

  return (
    <div className='chatEvents'>
      
      {chatItems}
    </div>
  )
}

// class ChatEvents extends React.Component {
//   render () {
//     const props = this.props;
//     return (
//       <div>
        
//       </div>
//     )
//   }
// }

export default ChatEvents;