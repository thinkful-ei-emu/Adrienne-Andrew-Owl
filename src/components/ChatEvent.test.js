import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatEvents from './ChatEvents';
import participants from '../data/participantData';
import chatEvents from '../data/chatEventsData';

// smoke test; failing time
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatEvents
    time={0}
    participants={[]}
    chatEvents={[]}
    />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// screenshot test; failing filter
it('renders this UI as expected', () => {
  const appTest = renderer.create(<ChatEvents
    time={chatEvents[0].time}
    participants={participants}
    chatEvents={chatEvents}
    />).toJSON();
  // Check whether it matches the previous snapshot
  // Stored in __snapshots__/App.test.js.snap
  expect(appTest).toMatchSnapshot(); 
});