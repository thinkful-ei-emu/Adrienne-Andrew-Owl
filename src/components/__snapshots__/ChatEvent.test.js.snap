// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders this UI as expected 1`] = `<div />`;
